firestore/integration_test_internal/src/query_snapshot_test.cc

```diff
@@ -30,5 +31,42 @@ TEST_F(QuerySnapshotTest, Assignment) {
 
 #endif  // defined(__ANDROID__)
 
+TEST_F(QuerySnapshotTest, TestCanQueryWithAndWithoutDocumentKey) {
+  CollectionReference collection =
+      Collection({{"a", {{"k", FieldValue::String("a")}}},
+                  {"b", {{"k", FieldValue::String("b")}}},
+                  {"c", {{"k", FieldValue::String("c")}}}});
+  QuerySnapshot snapshot1 = ReadDocuments(collection.Limit(2));
```

Comment:
Can you also include a default-constructed `QuerySnapshot` into the comparisons?

Reply:
Done